### PR TITLE
Hw 24(2)

### PR DESCRIPTION
**Pull request by glones**

### CODE REVIEW

File: src/main/java/com/internet/shop/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    private UserDao userDao;
+
+    @Override
+    public User create(User user) {
+        return userDao.create(user);
+    }
+
+    @Override
+    public User get(Long id) {
+        return userDao.get(id).orElseThrow();
```
Comment (andromash) [inline src/main/java/com/internet/shop/service/impl/UserServiceImpl.java:22]: better use userDao.get(id).get(). The last get will throw NoSuchElementException anyway. Try to change it in all service implementation

**Review by andromash — CHANGES_REQUESTED** [review]
well done!

File: src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    @Override
+    public List<Order> getUserOrders(Long userId) {
+        return orderDao.getOrdersOfUser(userId);
+    }
+
+    @Override
+    public Order get(Long id) {
+        return orderDao.get(id).orElseThrow();
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java:35]:
```suggestion
        return orderDao.get(id).get();
```
by default get() method throw NoSuchElementExcep

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    @Override
+    public void clear(ShoppingCart shoppingCart) {
+        shoppingCart.setProducts(new ArrayList<>());
+        shoppingCartDao.update(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart getByUserId(Long userId) {
+        return shoppingCartDao.getByUser(userId).orElseThrow();
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java:42]: the same

**Review by Sofasmile — COMMENTED** [review]
good job

File: src/main/java/com/internet/shop/service/impl/UserServiceImpl.java (hunk 1 above)
Comment (glones) [reply src/main/java/com/internet/shop/service/impl/UserServiceImpl.java]: done

File: src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java (hunk 2 above)
Comment (glones) [reply src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java]: done

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java (hunk 3 above)
Comment (glones) [reply src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]: done

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**